From d1ded2f5e47982e21e9f42a76fa0a7f2581d7798 Mon Sep 17 00:00:00 2001
From: Fugang Duan <B38611@freescale.com>
Date: Wed, 23 May 2012 14:21:56 +0800
Subject: [PATCH 321/505] ENGR00211144 SPDC : modify waveform update mode for
 test case.

- Because mode_4 only use for handwriting mode, and
  mode_3 use for power off mode, so use mode_2 for DU
  and A2, mode_1 for GU:
  mode_A2 = mode_2, mode_du = mode_2;
  mode_gu = mode_1;
- Modify the padding image in "Auto-update" case.

Signed-off-by: Fugang Duan  <B38611@freescale.com>
---
 test/mxc_fb_test/mxc_spdc_fb_test.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/test/mxc_fb_test/mxc_spdc_fb_test.c b/test/mxc_fb_test/mxc_spdc_fb_test.c
index 6d115b1..8c15337 100644
--- a/test/mxc_fb_test/mxc_spdc_fb_test.c
+++ b/test/mxc_fb_test/mxc_spdc_fb_test.c
@@ -64,10 +64,10 @@ extern "C"{
 #define BUFFER_OVERLAY		1
 
 #define WAVEFORM_MODE_INIT	0x0	/* Screen goes to white (clears) */
-#define WAVEFORM_MODE_DU	0x4	/* Grey->white/grey->black */
+#define WAVEFORM_MODE_DU	0x2	/* Grey->white/grey->black */
 #define WAVEFORM_MODE_GC16	0x1	/* High fidelity (flashing) */
 #define WAVEFORM_MODE_GC4	0x2	/* Lower fidelity */
-#define WAVEFORM_MODE_A2	0x4	/* Fast black/white animation */
+#define WAVEFORM_MODE_A2	0x2	/* Fast black/white animation */
 
 #define SPDC_STR_ID		"imx_spdc_fb"
 
@@ -780,7 +780,7 @@ static int test_auto_update(void)
 	printf("Blank screen\n");
 	memset(fb, 0x0, screen_info.xres_virtual*screen_info.yres*screen_info.bits_per_pixel/8);
 	update_to_display(0, 0, screen_info.xres, screen_info.yres,
-		WAVEFORM_MODE_DU, TRUE, 0);
+		WAVEFORM_MODE_INIT, TRUE, 0);
 
 	printf("Change to auto-update mode\n");
 	auto_update_mode = AUTO_UPDATE_MODE_AUTOMATIC_MODE;
@@ -792,16 +792,16 @@ static int test_auto_update(void)
 
 	printf("Auto-update 1st 5 lines\n");
 	for (i = 0; i < 5; i++) {
-		memset(fb, 0x00, screen_info.xres_virtual*5*screen_info.bits_per_pixel/8);
+		memset(fb, 0xFF, screen_info.xres_virtual*5*screen_info.bits_per_pixel/8);
 	}
 
 	sleep(6);
 
 	printf("Auto-update middle and lower stripes\n");
 	for (i = 0; i < 5; i++)
-		memset(fb + screen_info.xres_virtual*300*screen_info.bits_per_pixel/8/2, 0x00, screen_info.xres_virtual*5*screen_info.bits_per_pixel/8);
+		memset(fb + screen_info.xres_virtual*300*screen_info.bits_per_pixel/8/2, 0xFF, screen_info.xres_virtual*5*screen_info.bits_per_pixel/8);
 	for (i = 0; i < 5; i++)
-		memset(fb + screen_info.xres_virtual*500*screen_info.bits_per_pixel/8/2, 0x00, screen_info.xres_virtual*5*screen_info.bits_per_pixel/8);
+		memset(fb + screen_info.xres_virtual*500*screen_info.bits_per_pixel/8/2, 0xFF, screen_info.xres_virtual*5*screen_info.bits_per_pixel/8);
 
 	sleep(6);
 	printf("Auto-update blank screen\n");
-- 
1.8.0

